### PR TITLE
Fix: Correct Data Types for Setpoint Description, Selector, and HVAC

### PR DESCRIPTION
**Pull request by daviddsapir**
This commit addresses the data type issues in the `Setpoint` and `HVAC` model messages, specifically for `SetpointDescriptionDataType`, `SetpointDescriptionListDataSelectorsType`, and `HvacSystemFunctionSetpointRelationDataType`. It also corrects invalid types in `function_data_factory.go` for `FunctionTypeHvacOperationModeDescriptionListData` and `FunctionTypeHvacSystemFunctionDescriptionListData`.

Changes include:
1. Updating `SetpointId` in `hvacSystemFunctionSetpointRelationListData` to be a list, as specified in the EEBus SPINE Technical Specification Resource Specification.
2. Correcting data types for fields in `SetpointDescriptionDataType` and `SetpointDescriptionListDataSelectorsType` also according to the Resource Specification.

Before this commit, sending `FunctionTypeHvacOperationModeDescriptionListData` and `FunctionTypeHvacSystemFunctionOperationModeRelationListData` failed due to invalid data types provided to `createFunctionData` in the factory.

Additionally, the `SetpointId` field needed to be a list, not a single value.  According to the specification, an operation mode can have multiple setpoints (up to four), such as for the "auto" operation mode.

Sending `FunctionTypeSetpointDescriptionListData` also failed due to incorrect field data types.

With these fixes, requests for `FunctionTypeSetpointDescriptionListData`, `FunctionTypeHvacOperationModeDescriptionListData`, and `FunctionTypeHvacSystemFunctionOperationModeRelationListData` now work correctly.

### 4.3.12.6 hvacSystemFunctionSetpointRelationListData
image

### 4.3.23.6 setpointDescriptionListData (for `SetpointDescriptionDataType`)
image

### 5.3.21.4.3 Available selectors (for `SetpointDescriptionListDataSelectorsType`)
image

### CODE REVIEW

**Comment by DerAndereAndi** [comment]
Thanks for these findings and fixes. Very much appreciated!

File: model/setpoint.go [hunk 1]
```diff
@@ -64,14 +64,14 @@ type SetpointConstraintsListDataSelectorsType struct {
...
-	MeasurementId *SetpointIdType   `json:"measurementId,omitempty"`
-	TimeTableId   *SetpointIdType   `json:"timeTableId,omitempty"`
-	SetpointType  *SetpointTypeType `json:"setpointType,omitempty"`
-	Unit          *ScaledNumberType `json:"unit,omitempty"`
-	ScopeType     *ScaledNumberType `json:"scopeType,omitempty"`
-	Label         *LabelType        `json:"label,omitempty"`
-	Description   *DescriptionType  `json:"description,omitempty"`
+	SetpointId    *SetpointIdType        `json:"setpointId,omitempty" eebus:"key"`
+	MeasurementId *SetpointIdType        `json:"measurementId,omitempty"`
+	TimeTableId   *SetpointIdType        `json:"timeTableId,omitempty"`
```
Comment (DerAndereAndi) [inline model/setpoint.go]: `MeasurementId` and `TimeTableId` are documented as being foreign identifiers. Therefor the also need to get the tag annotation `eebus:"key"`, like `SetpointId` has

**Review by DerAndereAndi — CHANGES_REQUESTED** [review]

File: model/setpoint.go (hunk 1 above)
Comment (daviddsapir) [reply model/setpoint.go]: Fixed. Thanks for noticing!
I added `eebus:"key"` to both foreign identifiers, `MeasurementId` and `TimeTableId`, in the `SetpointDescriptionDataType` class, just like it is for `SetpointId`.

**Review by DerAndereAndi — APPROVED** [review]

**Comment by DerAndereAndi** [comment]
I think the failing test code also needs to include the foreign identifiers in the sample data structures for the merge to work correctly. Could you take a look at this?

**Comment by daviddsapir** [comment]
I’ve fixed the failing test

**Comment by DerAndereAndi** [comment]
Awesome, thank you very much!

**Pull request merged**